components/eamxx/src/physics/shoc/eamxx_shoc_process_interface.hpp

```diff
@@ -138,8 +138,9 @@ class SHOCMacrophysics : public scream::AtmosphereProcess
 
         // Dry static energy
         shoc_s(i,k) = PF::calculate_dse(T_mid(i,k),z_mid(i,k),phis(i));
+
+        if (k+1 == nlev_packs) zi_grid(i,nlevi_v)[nlevi_p] = 0;
       });
-      zi_grid(i,nlevi_v)[nlevi_p] = 0;
```

Comment:
Placing a team_barrier before this line would also work.

Reply:
I'm puzzled. Why was this line buggy? All threads would execute that line. Albeit being unnecessary (so I'm ok with the change), it would not be incorrect, since they all set the same value, and there was a barrier right after.

Reply:
I agree. If the previous kernel was over nlevi rather than nlev, I could make an argument about memory consistency. But since it's over nlev, I can think of any explanation. Maybe this fix is actually working around a compiler-side issue.

The evidence for this specific line being the issue is that the _sfc quantities a few lines below this code block were all wrong. It's possible there's some other bug I'm not seeing that this fix handles. If so, that bug must affect these _sfc quantities.

I'll weaken my suggestion that other C++ devs study this PR since it may just be working around a compiler-side problem.